lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -135,3 +137,48 @@ export async function detectPnpmWorkspaces(
     }
   }
 }
+
+export async function getPnpmShrinkwrap(filePath: string): Promise<LockFile> {
+  // TODO #7154
+  const pmpmLockRaw = (await readLocalFile(filePath, 'utf8'))!;
+  try {
+    const lockParsed = yaml.load(pmpmLockRaw) as Record<string, any>;
+    logger.debug({ lockParsed }, 'pnpm lockfile parsed');
+    const lockedVersions: Record<string, string> = {};
+    const packagePathRegex = regEx(
+      /\/(?<packageName>.*)(?<version>\d\.\d\.\d.*)/
```

Comment:
are you sure versions are always semver compatible?

```suggestion
      /^\/(?<packageName>.*)(?:@|/)(?<version>\d\.\d\.\d[^/@]*)\/$/
```
maybe something like that?

Reply:
Why excluding `@`, `/` and adding a slash at the end?

Reply:
>are you sure versions are always semver compatible?
I thought they will be since pnpm is widely used.

Reply:
1. because you then don't need to strip it from package name later
2. ups, was a fault assumption

Reply:
1. I meant the ones at the end after version

Reply:
it's negative group see the `^`

Reply:
updated suggestion